### PR TITLE
Poc2

### PR DESCRIPTION
**Pull request by marius-se**
This PR focusses on the the first step of the registration.

### CODE REVIEW

File: Sources/App/routes.swift [hunk 1]
```diff
@@ -34,102 +34,107 @@ func routes(_ app: Application) throws {
...
         let username = try req.query.get(String.self, at: "username")
         guard let user = try await User.query(on: req.db).filter(\.$username == username).first() else {
             throw Abort(.unauthorized)
         }
-        let challenge = [UInt8].random(count: 32).base64
-        let encodedChallenge = challenge.replacingOccurrences(of: "+", with: "-")
-            .replacingOccurrences(of: "/", with: "_")
-            .replacingOccurrences(of: "=", with: "")
+        let challenge = try req.webAuthn.generateChallenge()
+        let encodedChallenge = challenge.base64URLEncodedString()
```
Comment (0xTim) [inline Sources/App/routes.swift:102]: This should probably just be `req.webAuthn.generateChallengeString()`

File: Sources/App/routes.swift [hunk 2]
```diff
@@ -34,102 +34,107 @@ func routes(_ app: Application) throws {
...
             // We need to convert the IDs to base64 encoded from base64 URL encoded
-            var id = credential.id!.replacingOccurrences(of: "-", with: "+").replacingOccurrences(of: "_", with: "/")
+            var id = String.base64(fromBase64URLEncoded: credential.id!)
             while id.count % 4 != 0 {
                 id = id.appending("=")
             }
             return WebAuthnCredential(id: id, publicKey: credential.publicKey, userID: credential.$user.id)
         }
-        return StartAuthenticateResponse(challenge: challenge, credentials: credentials)
+        return StartAuthenticateResponse(challenge: challenge.base64String(), credentials: credentials)
```
Comment (0xTim) [inline Sources/App/routes.swift:114]: I guess we can probably roll these into helpers too

**Review by 0xTim — APPROVED** [review]
Looks great! Added a couple of suggestions for things we can roll into helpers

File: Sources/App/routes.swift (hunk 2 above)
Comment (marius-se) [reply Sources/App/routes.swift:114]: yep that endpoint i didn't touch yet

File: Sources/App/routes.swift (hunk 1 above)
Comment (marius-se) [reply Sources/App/routes.swift:102]: 👍🏼

**Pull request merged**